18-filter-in-trees.md

```diff
@@ -0,0 +1,86 @@
+---
+layout: page
+title: Second steps in LHCb
+subtitle: Reuse particles from a decay tree
+minutes: 10
+---
+
+> ## Learning Objectives {.objectives}
+>
+> * Learn how to extract particles from a decay tree
+> * Build a new particle from the extracted particles
+
+Sometimes we want to extract a portion of the decay tree in order to build a different decay.
+To do that, we need to put the particles we're interested in in a new container so they can afterwards be used as inputs to a `CombineParticles` instance (as we saw in [the selection framework lesson](https://lhcb.github.io/second-analysis-steps/01-building-decays.html)).
+To achieve this we can use the `FilterInTrees` algorithm, a simple variation of `FilterDesktop` ([doxygen](https://lhcb-release-area.web.cern.ch/LHCb-release-area/DOC/hlt/latest_doxygen/de/d8e/class_filter_in_trees.html)).
+
+Let's start from the example in [the selection framework lesson](https://lhcb.github.io/second-analysis-steps/01-building-decays.html) and let's check that the $\pi^-$ child of the $D^0$ does not come from a $\rho\to\pi^+\pi^-$.
+To do that, we have to extract the $\pi^-$ from `([D0 -> pi- K+]CC)` and combine it with all pions in `Phys/StdAllNoPIDsPions/Particles`.
+
+Using `FilterInTrees` is done in the same way we would use `FilterDesktop`:
+
+```python
+from Configurables import FilterInTrees
+from PhysSelPython.Wrappers import Selection, DataOnDemand
+
+decay_tree_location = '/Event/AllStreams/Phys/D2hhCompleteEventPromptDst2D2RSLine/Particles'
+d0_from_dst = FilterInTrees('d0_from_dst_filter', Code="DECTREE('[Charm -> pi- K+]CC')")
+d0_from_dst_sel = Selection("d0_from_dst_sel",
+                            Algorithm=d0_from_dst,
+                            RequiredSelections=[AutomaticData(Location=decay_tree_location)])
+pions_from_d0 = FilterInTrees('pions_from_d0_filter',Code="('pi+' == ABSID)")
+pions_from_d0_sel = Selection("pions_from_d0_sel",
+                              Algorithm=pions_from_d0,
+                              RequiredSelections=[d0_from_dst_sel])
```

Comment:
Unfortunately, the `DECTREE` thing doesn't work (I tried) because `FilterInTree` loops over the particles and it doesn't allow to mark particles. 

I'll fix the decay, I thought I had taken it from the first analysis steps... We should check it there just in case.